### PR TITLE
jit(arm64): support for and, or, and xor

### PR DESCRIPTION
**Pull request by summerwind**
Hello, The JIT support for Arm64 is really nice!
This PR adds support of `and`, `or`, and `xor` instruction for Arm64 JIT.
This might be a part of #187.

I have implemented this with reference to the x86 JIT code.
If you have any concerns about this, I would appreciate your review comments!

### CODE REVIEW

File: wasm/jit/jit_arm64.go [hunk 1]
```diff
@@ -771,16 +771,86 @@ func (c *arm64Compiler) compileRem(o *wazeroir.OperationRem) error {
...
+		return err
+	}
+
+	// At this point, at least one of x1 or x2 registers is non zero.
+	// Choose the non-zero register as destination.
+	var destinationReg int16 = x1.register
+	if isZeroRegister(x1.register) {
+		destinationReg = x2.register
+	}
+
```
Comment (mathetake) [inline wasm/jit/jit_arm64.go:850]: could you choose either `AEOR` or `AEORW` according to o.Type (size of integer)?

**Review by mathetake — COMMENTED** [review]
Awesome! Thank you for this, one small fix and then good to go!

**Review by mathetake — APPROVED** [review]
LGTM!

**Comment by mathetake** [comment]
Thank you for your contribution (again)! 👯

**Comment by summerwind** [comment]
Thank you!

**Comment by codefromthecrypt** [comment]
it is exciting and motivating to see you again @summerwind 🙇

**Pull request merged**